dot/sync/test_helpers.go

```diff
@@ -51,3 +70,201 @@ func (bp *mockBlockProducer) Resume() error {
 }
 
 func (bp *mockBlockProducer) SetRuntime(_ runtime.Instance) {}
+
+type mockFinalityGadget struct{}
+
+func (m mockFinalityGadget) VerifyBlockJustification(_ []byte) error {
+	return nil
+}
+
+// NewTestSyncer ...
+func NewTestSyncer(t *testing.T) *Service {
+	wasmer.DefaultTestLogLvl = 3
+
+	cfg := &Config{}
+	testDatadirPath, _ := ioutil.TempDir("/tmp", "test-datadir-*")
+	stateSrvc := state.NewService(testDatadirPath, log.LvlInfo)
+	stateSrvc.UseMemDB()
+
+	gen, genTrie, genHeader := newTestGenesisWithTrieAndHeader(t)
+	err := stateSrvc.Initialise(gen, genHeader, genTrie)
+	require.NoError(t, err)
+
+	err = stateSrvc.Start()
+	require.NoError(t, err)
+
+	if cfg.BlockState == nil {
+		cfg.BlockState = stateSrvc.Block
+	}
+
+	if cfg.StorageState == nil {
+		cfg.StorageState = stateSrvc.Storage
+	}
+
+	if cfg.Runtime == nil {
+		// set state to genesis state
+		genState, err := rtstorage.NewTrieState(genTrie) //nolint
+		require.NoError(t, err)
+
+		rtCfg := &wasmer.Config{}
+		rtCfg.Storage = genState
+		rtCfg.LogLvl = 3
+
+		instance, err := wasmer.NewRuntimeFromGenesis(gen, rtCfg) //nolint
+		require.NoError(t, err)
+		cfg.Runtime = instance
+	}
+
+	if cfg.TransactionState == nil {
+		cfg.TransactionState = stateSrvc.Transaction
+	}
+
+	if cfg.Verifier == nil {
+		cfg.Verifier = &mockVerifier{}
+	}
+
+	if cfg.LogLvl == 0 {
+		cfg.LogLvl = log.LvlDebug
+	}
+
+	if cfg.FinalityGadget == nil {
+		cfg.FinalityGadget = &mockFinalityGadget{}
+	}
+
+	syncer, err := NewService(cfg)
+	require.NoError(t, err)
+	return syncer
+}
+
+func newTestGenesisWithTrieAndHeader(t *testing.T) (*genesis.Genesis, *trie.Trie, *types.Header) {
+	gen, err := genesis.NewGenesisFromJSONRaw("../../chain/gssmr/genesis.json")
+	require.NoError(t, err)
+
+	genTrie, err := genesis.NewTrieFromGenesis(gen)
+	require.NoError(t, err)
+
+	genesisHeader, err := types.NewHeader(common.NewHash([]byte{0}), genTrie.MustHash(), trie.EmptyHash, big.NewInt(0), types.Digest{})
+	require.NoError(t, err)
+	return gen, genTrie, genesisHeader
+}
+
+// BuildBlock ...
+func BuildBlock(t *testing.T, srv *Service, parent *types.Header, ext types.Extrinsic) *types.Block {
```

Comment:
I don't think this needs to be in the `sync` package as it doesn't rely on anything in it, it would be better placed in `lib/utils` or something potentially. also now that `lib/babe` has a `BlockBuilder` type you could instantiate that to build a block instead

Reply:
`BuildBlock ` is used in a couple of tests in `syncer_test.go`. I tried moving it on `lib/utils` but I am getting cyclic dependency.